js/src/views/Signer/store.js

```diff
@@ -49,8 +49,13 @@ export default class SignerStore {
   }
 
   @action unsubscribe () {
-    if (this._timeoutId) {
-      clearTimeout(this._timeoutId);
+    const self = this;
+
+    if (self.subscribed) {
+      if (!self.subscriptionId) {
```

Comment:
Ah, I understand your design now and how nice that is. Sorry I didn't fully understand your first comment.